### PR TITLE
Add initial documentation to support blog post

### PR DESCRIPTION
**Pull request by tbekolay**

### CODE REVIEW

**Review by drasmuss — COMMENTED** [review]
Pushed a fixup with a few changes. Updated the docs requirements in setup.py and added the docs build to CI. Then I moved the detailed installation instructions out of the README, because the tabs don't get rendered when you're just looking at the file directly, and I thought that made the installation instructions look a bit intimidating. I also separated the TensorFlow and Documentation installation instructions into different sections (since a user could want to install TensorFlow without caring about running the examples).

File: README.rst [hunk 1]
```diff
@@ -2,7 +2,113 @@
...
+
+      1. Download the `Miniconda installer <https://docs.conda.io/en/latest/miniconda.html#latest-miniconda-installer-links>`_
+         for your OS and follow the installation instructions on that page.
+
+
+      2. Open a terminal or command-line prompt and ensure conda is up to date.
+         Configure it to use conda-forge by default::
+
+           conda update conda
+           conda config --add channels conda-forge
```
[the remark this answers is not in the document]
Comment (tbekolay) [reply README.rst]: conda-forge is always an option; my understanding is that this moves it to the top of the list so it gets used first. At least, this is what the conda-forge docs say to do: https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge

But I haven't tested it, I'll test it now.

[the remark this answers is not in the document]
Comment (drasmuss) [reply README.rst]: Ah yeah you're right, for some reason I'd thought that when I was wrestling with all the tensorflow channel stuff I ended up having to use something other than `--add`, but I was trying all kinds of stuff there so who knows what was going on.

**Comment by drasmuss** [comment]
Docs build is failing because the `gh-pages-release` branch doesn't exist yet, but once that is there I think it should work.

**Comment by tbekolay** [comment]
I made an empty `gh-pages-release` branch so the docs build should succeed, and pushed a fixup with the changes we talked about this morning. I think this is all good to go now, so feel free to squash / merge whenever @drasmuss

**Pull request merged**